File: Resueltos_por_competidor/Jacobo_Segura/Crochet.java
import java.util.Scanner;

public class Crochet {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        String startDay = sc.next();
        String startTime = sc.next();

        String endDay = sc.next();
        String endTime = sc.next();

        int start = convertir(startDay, startTime);
        int end = convertir(endDay, endTime);

        int diff = end - start;

        if (diff <= 0) {
            diff += 7 * 24 * 60;
        }

        int days = diff / (24 * 60);
        diff %= (24 * 60);

        int hours = diff / 60;
        int minutes = diff % 60;

        StringBuilder ans = new StringBuilder();

        int count = 0;
        if (days > 0) count++;
        if (hours > 0) count++;
        if (minutes > 0) count++;

        int printed = 0;

        if (days > 0) {
            ans.append(days).append(days == 1 ? " day" : " days");
            printed++;
        }

        if (hours > 0) {
            if (printed > 0) {
                if (count == 3) {
                    ans.append(", ");
                } else {
                    ans.append(" and ");
                }
            }
            ans.append(hours).append(hours == 1 ? " hour" : " hours");
            printed++;
        }

        if (minutes > 0) {
            if (printed > 0) {
                if (count == 3) {
                    ans.append(", ");
                } else {
                    ans.append(" and ");
                }
            }
            ans.append(minutes).append(minutes == 1 ? " minute" : " minutes");
        }

        System.out.println(ans);
    }

    static int convertir(String day, String time) {
        int d = 0;
        switch (day) {
            case "Mon": d = 0; break;
            case "Tue": d = 1; break;
            case "Wed": d = 2; break;
            case "Thu": d = 3; break;
            case "Fri": d = 4; break;
            case "Sat": d = 5; break;
            case "Sun": d = 6; break;
        }

        int hour = Integer.parseInt(time.substring(0, 2));
        int minute = Integer.parseInt(time.substring(3));

        return d * 24 * 60 + hour * 60 + minute;
    }
}

File: Resueltos_por_competidor/German_Scioville/Crochet.cpp
/*
 * Autor: Scioville
 * Problema: Crochet Competition (https://open.kattis.com/problems/crochetcompetition)
 * Juez online: Katiis
 * Veredicto: accepted  
 * After all.Not all good times must come to an end.
 * */
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int d, f, h, m, x, y, auxd, auxh, auxm, aux, aux2, t;

    string date1,date2,time1,time2,dia, hour, minuto;
    d = 0;
    f = 0;
    dia = "days";
    hour = "hours";
    minuto = "minutes";

    cin >> date1;
    cin >>time1;

    cin >> date2;
    cin >>time2;

    h = stoi(time1.substr(0,2));
    m = stoi(time1.substr(3,2));

    x = stoi(time2.substr(0,2));
    y = stoi(time2.substr(3,2));

    string day[] = { "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun" };

    for (int i = 0; i < 7; i++)
    {
        if(date1==day[i])
        {
            d = i+1;
        }

        if(date2==day[i])
        {
            f = i+1;
        }
    }

    aux = (d * 1440) + (h * 60) + m;
    aux2 = (f * 1440) + (x * 60) + y;

    t = aux2 - aux;
    if (t < 0)
    {
        t += 7 * 1440;
    }
    else if (t == 0)
    {
        t = 7 * 1440;
    }

    auxd = t / 1440;
    t %= 1440;

    auxh = t / 60;
    auxm = t % 60;

    if (auxd == 1)
    {
        dia = "day";
    }

    if (auxh == 1)
    {
        hour = "hour";
    }
    if (auxm == 1)
    {
        minuto = "minute";
    }

    if (auxd != 0 && auxh != 0 && auxm != 0)
    {
        cout << auxd << " " << dia << ", " << auxh << " " << hour << ", " << auxm << " " << minuto << endl;
    }
    else if (auxd == 0 && auxh != 0 && auxm != 0)
    {
        cout << auxh << " " << hour << " and " << auxm << " " << minuto << endl;
    }
    else if (auxd == 0 && auxh == 0 && auxm != 0)
    {
        cout << auxm << " " << minuto << endl;
    }
    else if (auxd != 0 && auxh == 0 && auxm == 0)
    {
        cout << auxd << " " << dia << endl;
    }
    else if (auxd != 0 && auxh != 0 && auxm == 0)
    {
        cout << auxd << " " << dia << " and " << auxh << " " << hour << endl;
    }
    else if (auxd != 0 && auxh == 0 && auxm != 0)
    {
        cout << auxd << " " << dia << " and " << auxm << " " << minuto << endl;
    }
    else if (auxd == 0 && auxh != 0 && auxm == 0)
    {
        cout << auxh << " " << hour << endl;
    }
    else if (auxd == 0 && auxh == 0 && auxm == 0)
    {
        cout << "0 minutes" << endl;
    }
}
